src/components/counters/Counter.tsx

```diff
@@ -83,21 +141,28 @@ const Counter = ({
   children,
   className,
   size,
-  color = 'red-60',
+  color = 'red',
+  type = 'solid',
   withAnimation,
   'aria-label': ariaLabel,
   ...props
 }: CounterPropsType) => {
+  const backgroundColor =
+    type === 'solid' ? COLORS_SOLID_MAP[color] : COLORS_LIGHT_MAP[color];
   const counterClass = cx(
     'sg-counter',
     {
       [`sg-counter--${String(size)}`]: size,
-      [`sg-counter--${String(color)}`]: color,
+      [`sg-counter--${String(backgroundColor)}`]: backgroundColor,
       'sg-counter--with-animation': withAnimation,
       'sg-counter--with-icon': icon,
     },
     className
   );
+
+  const textColor =
+    type === 'solid' ? SOLID_COLOR_TEXT_MAP[color] : TEXT_COLOR['text-black'];
```

Comment:
```suggestion
    type === 'solid' ? SOLID_COLOR_TEXT_MAP[color] : 'text-black';
```

you don't need TEXT_COLOR since you're not mapping/or iterating over anything here. Typescript will take care if the literal you've used is corect.